### PR TITLE
Close Circular and Schema Version Dropdowns When Clicking Away 

### LINKED ISSUE #2345: Automatically close version dropdowns for Circulars and Schema Browser when clicking outside
# Description

Automatically close the version dropdowns on the Circulars and Schema Browser pages when clicking outside them.

# Acceptance criteria

Employ usehooks-ts to automatically close the dropdowns when clicking outside.

### PR DESCRIPTION
**Pull request by tylerbarna**
# Description
This adds the functionality from #1989 to the other two dropdown menus located in the repo at the moment: the version dropdown menus for the Schema browser and any circulars with multiple versions. 

The method for determining if the dropdown content is shown or hidden for these has also been updated to match the other two dropdowns.

# Related Issue(s)

Resolves #2345 

# Testing
Schema Browser Dropdown:
1. Navigate to Docs page
2. Select 'Schema Browser' in menu
3. Click on Version dropdown in upper right-hand corner to expand the dropdown content
4. Clicking on the button again should collapse the dropdown content, and clicking elsewhere on the page while the dropdown is expanded should also collapse it

Circulars Dropdown:
1. Navigate to a Circular with multiple versions (example: 34776)
2. Click on Version dropdown in upper right-hand corner to expand the dropdown content
3. Clicking on the button again should collapse the dropdown content, and clicking elsewhere on the page while the dropdown is expanded should also collapse it

### CODE REVIEW

File: app/routes/circulars.$circularId.($version)/route.tsx [hunk 1]
```diff
@@ -154,10 +156,13 @@ function CircularsHistory({
   circular: number
   history?: number[]
 }) {
-  const [showVersions, setShowVersions] = useState<boolean>(false)
-
+  const ref = useRef<HTMLDivElement>(null)
+  const [showVersions, setShowVersions] = useState(false)
```
Comment (tylerbarna) [inline app/routes/circulars.$circularId.($version)/route.tsx:160]:
```suggestion
  const [showContent, setShowContent] = useState(false)
```

File: app/routes/circulars.$circularId.($version)/route.tsx [hunk 2]
```diff
@@ -154,10 +156,13 @@ function CircularsHistory({
...
+  useOnClickOutside(ref, () => {
+    setShowVersions(false)
```
Comment (tylerbarna) [inline app/routes/circulars.$circularId.($version)/route.tsx:162]:
```suggestion
    setShowContent(false)
```

File: app/routes/circulars.$circularId.($version)/route.tsx [hunk 3]
```diff
@@ -166,29 +171,31 @@ function CircularsHistory({
...
-                  </Link>
-                </div>
-              ))}
-          </CardBody>
-        </DetailsDropdownContent>
-      )}
-    </>
+      <DetailsDropdownContent
+        className={classNames('maxw-card-xlg', {
+          'display-none': !showVersions,
```
Comment (tylerbarna) [inline app/routes/circulars.$circularId.($version)/route.tsx:176]:
```suggestion
          'display-none': !showContent,
```

File: app/routes/circulars.$circularId.($version)/route.tsx [hunk 4]
```diff
@@ -166,29 +171,31 @@ function CircularsHistory({
...
+        })}
+      >
+        <CardBody>
+          <Link
+            onClick={() => setShowVersions(!showVersions)}
```
Comment (tylerbarna) [inline app/routes/circulars.$circularId.($version)/route.tsx:181]:
```suggestion
            onClick={() => setShowContent(!showContent)}
```

File: app/routes/circulars.$circularId.($version)/route.tsx [hunk 5]
```diff
@@ -166,29 +171,31 @@ function CircularsHistory({
...
+            to={`/circulars/${circular}`}
+          >
+            Latest
+          </Link>
+          {history &&
+            history.map((version) => (
+              <div key={version}>
+                <Link
+                  onClick={() => setShowVersions(!showVersions)}
```
Comment (tylerbarna) [inline app/routes/circulars.$circularId.($version)/route.tsx:190]:
```suggestion
                  onClick={() => setShowContent(!showContent)}
```

File: app/routes/docs_._schema-browser.schema.($version).$/route.tsx [hunk 6]
```diff
@@ -125,10 +126,14 @@ function VersionSelector({
   versions: { name: string | null; ref: string }[]
   path: string
 }) {
+  const ref = useRef<HTMLDivElement>(null)
   const [showVersions, setShowVersions] = useState(false)
```
Comment (tylerbarna) [inline app/routes/docs_._schema-browser.schema.($version).$/route.tsx:130]:
```suggestion
  const [showContent, setShowContent] = useState(false)
```

File: app/routes/docs_._schema-browser.schema.($version).$/route.tsx [hunk 7]
```diff
@@ -125,10 +126,14 @@ function VersionSelector({
...
+  useOnClickOutside(ref, () => {
+    setShowVersions(false)
```
Comment (tylerbarna) [inline app/routes/docs_._schema-browser.schema.($version).$/route.tsx:132]:
```suggestion
    setShowContent(false)
```

File: app/routes/docs_._schema-browser.schema.($version).$/route.tsx [hunk 8]
```diff
@@ -137,29 +142,31 @@ function VersionSelector({
...
-                </Link>
-              </div>
-            ))}
-          </CardBody>
-        </DetailsDropdownContent>
-      )}
-    </>
+      <DetailsDropdownContent
+        className={classNames('maxw-card-xlg', {
+          'display-none': !showVersions,
```
Comment (tylerbarna) [inline app/routes/docs_._schema-browser.schema.($version).$/route.tsx:147]:
```suggestion
          'display-none': !showContent,
```

File: app/routes/docs_._schema-browser.schema.($version).$/route.tsx [hunk 9]
```diff
@@ -137,29 +142,31 @@ function VersionSelector({
...
+          <h3>Versions</h3>
+        </CardHeader>
+        <CardBody className="padding-y-0">
+          {versions.map(({ name, ref }) => (
+            <div key={ref}>
+              <Link
+                className="usa-link"
+                to={`/docs/schema/${ref}/${path}`}
+                onClick={() => {
+                  setShowVersions(false)
```
Comment (tylerbarna) [inline app/routes/docs_._schema-browser.schema.($version).$/route.tsx:160]:
```suggestion
                  setShowContent(false)
```

**Review by tylerbarna — COMMENTED** [review]
These changes would result in them matching the defined variables of the other dropdowns more closely but increases the diff of this PR

File: app/routes/circulars.$circularId.($version)/route.tsx [hunk 10]
```diff
@@ -166,29 +171,31 @@ function CircularsHistory({
...
-              ))}
-          </CardBody>
-        </DetailsDropdownContent>
-      )}
-    </>
+      <DetailsDropdownContent
+        className={classNames('maxw-card-xlg', {
+          'display-none': !showVersions,
+        })}
+      >
```
Comment (lpsinger) [inline app/routes/circulars.$circularId.($version)/route.tsx:178]: Is it important that we switch from conditionally rendering this content to conditionally styling it as not displayed?

Comment (lpsinger) [reply app/routes/circulars.$circularId.($version)/route.tsx:178]: Yes, please switch it back.

Comment (tylerbarna) [reply app/routes/circulars.$circularId.($version)/route.tsx]: reverted in [652d205](https://github.com/nasa-gcn/gcn.nasa.gov/pull/2362/commits/652d2057aab6cb8169e82628ceb3e9a9edc7b9e9)

File: app/routes/circulars.$circularId.($version)/route.tsx (hunk 1 above)
Comment (lpsinger) [reply app/routes/circulars.$circularId.($version)/route.tsx:160]: It's your PR. Please just go ahead and make those changes.

**Review by lpsinger — APPROVED** [review]

**Comment by lpsinger** [comment]
Thank you!

**Pull request merged**